### PR TITLE
🧪 Complete test coverage for auth.ts functions

### PR DESCRIPTION
**Pull request by is0692vs**
🎯 **What:**
The task requested adding missing tests for OAuth-related functions in `packages/web/src/lib/auth.ts`, specifically noting `buildNotionRedirectUri` and `getAccessToken` which involve mocking request and response objects. I identified that numerous other helper functions in `auth.ts` were also lacking test coverage and implemented comprehensive unit tests for all of them.

📊 **Coverage:**
New test cases cover the following previously untested functions and logic paths in `packages/web/src/lib/auth.ts`:
- `buildNotionRedirectUri`: Mocked various `Headers` states (including `x-forwarded-proto`, `x-forwarded-host`, and fallbacks).
- `setAuthCookies`: Validated exact cookie options, including specific conditional logic for the `secure` flag based on request protocols.
- `isSecureRequest`: Extensively tested edge cases depending on request headers (`host`, `x-forwarded-proto`) versus `NODE_ENV` fallbacks.
- `getRefreshToken`, `getSelectedDatabaseId`, `getUserInfo`, `getNotionClient`, `isAuthenticated`.
- `setDatabaseCookie`, `setOauthStateCookie`, `createStateToken`.

✨ **Result:**
The test suite in `packages/web/src/lib/auth.test.ts` was expanded from 13 tests to 36 passing tests. The overall file code coverage for `auth.ts` has significantly increased, providing a robust safety net for all OAuth operations. Tested logic now handles and verifies environment edge cases seamlessly.

---
*PR created automatically by Jules for task [437842815290414522](https://jules.google.com/task/437842815290414522) started by @is0692vs*

<details open><summary><h3>Greptile Summary</h3></summary>

このPRは `auth.ts` の全公開関数を対象にテストを13件から36件へ拡充し、`getRefreshToken` の型ガードを `??` から `typeof ... === "string"` チェックへ修正しています。

- `buildNotionRedirectUri`・`setAuthCookies`・`isSecureRequest` 等の未テスト関数に対して、ヘッダーのエッジケースを含む網羅的なテストを追加。
- `getRefreshToken` の実装バグ（数値トークンを文字列として返す）を修正し、対応するリグレッションテストも追加。
- 以前のレビューで指摘されたインデント混在・`undefined` リクエストのモックパターンも合わせて修正済み。
</details>

<details open><summary><h3>Confidence Score: 5/5</h3></summary>

テスト追加と小さな実装バグ修正のみで、既存の認証ロジックには影響なし。安全にマージ可能。

変更内容はテストファイルの拡充と `getRefreshToken` の一行修正のみ。修正後の型ガードは `getAccessToken` と対称になっており、追加テストで動作が検証されている。

`auth.test.ts` の `setDatabaseCookie` テストブロックで環境変数クリーンアップの配置に注意。
</details>

<details open><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| packages/web/src/lib/auth.test.ts | 13件から36件へテストを大幅拡充。`setDatabaseCookie` テストで `vi.unstubAllEnvs()` が `afterEach` ではなく `it` ブロック末尾にあり、テスト失敗時に環境変数が汚染される恐れがある。`getRefreshToken` に不正クッキー値のテストケースが欠落している（`getAccessToken` には存在）。 |
| packages/web/src/lib/auth.ts | `getRefreshToken` の戻り値検証を `??` から `typeof ... === "string"` チェックに修正。`getAccessToken` と対称な安全なガードになった。 |

</details>

</details>

<h3>Flowchart</h3>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A[クッキーストア / リクエスト] --> B{getAccessToken / getRefreshToken}
    B --> C[unsealCookieValue]
    C --> D{typeof token === 'string'?}
    D -- Yes --> E[トークン文字列を返す]
    D -- No / null --> F[null を返す]

    G[リクエストヘッダー] --> H{isSecureRequest}
    H --> I{x-forwarded-proto あり?}
    I -- https --> J[secure: true]
    I -- http --> K[secure: false]
    I -- なし --> L{host が localhost?}
    L -- Yes --> K
    L -- No --> J
    H --> M{request なし?}
    M --> N{NODE_ENV === production?}
    N -- Yes --> J
    N -- No --> K

    J & K --> O[setAuthCookies / setDatabaseCookie / setOauthStateCookie]
```

<details><summary>Prompt To Fix All With AI</summary>

`````markdown
Fix the following 2 code review issues. Work through them one at a time, proposing concise fixes.

---

### Issue 1 of 2
packages/web/src/lib/auth.test.ts:462-502
`vi.unstubAllEnvs()` がテストケース本体の末尾で呼ばれており、アサーションが失敗した場合に実行されません。`afterEach` に移すことで、テスト失敗時も確実に環境変数が復元され、後続テストへの汚染を防げます。

```suggestion
	describe("setDatabaseCookie specific options tests", () => {
		let mockResponse: MockResponse;

		beforeEach(() => {
			mockResponse = createMockResponse();
		});

		afterEach(() => {
			vi.unstubAllEnvs();
		});

		it("should correctly set database cookie with default secure option based on NODE_ENV", () => {
			vi.stubEnv("NODE_ENV", "production");
			setDatabaseCookie(mockResponse, "db-123");
			expect(mockResponse.cookies.set).toHaveBeenCalledWith(
				DATABASE_ID_COOKIE,
				"db-123",
				{
					httpOnly: true,
					secure: true,
					sameSite: "lax",
					path: "/",
					maxAge: 2592000,
				},
			);
		});

		it("should correctly set database cookie with default secure option false based on NODE_ENV", () => {
			vi.stubEnv("NODE_ENV", "development");
			setDatabaseCookie(mockResponse, "db-123");
			expect(mockResponse.cookies.set).toHaveBeenCalledWith(
				DATABASE_ID_COOKIE,
				"db-123",
				{
					httpOnly: true,
					secure: false,
					sameSite: "lax",
					path: "/",
					maxAge: 2592000,
				},
			);
		});
	});
```

### Issue 2 of 2
packages/web/src/lib/auth.test.ts:337-378
`getRefreshToken` のテストに「不正な形式のクッキー値（例: `"invalid.cookie.value"`）を渡した場合に `null` を返す」ケースが抜けています。`getAccessToken` には同等のテスト（行 579–583）が存在しており、両関数は同一の `unsealCookieValue` を経由するため、一貫性のためにも追加が望ましいです。

`````

</details>

<sub>Reviews (2): Last reviewed commit: ["Address auth test review feedback"](https://github.com/hiroki-org/paper-tools/commit/c26621935469f5cc7625e88150dc43e172ac3cd5) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=32529102)</sub>

### CODE REVIEW

File: packages/web/src/lib/auth.test.ts [hunk 1]
```diff
@@ -1,199 +1,599 @@
...
+		it("should return null for malformed cookie value", () => {
+			expect(unsealCookieValue("not-a-valid-format")).toBeNull();
+			expect(unsealCookieValue("")).toBeNull();
+		});
+
+		it("should return null if payload is valid base64url but invalid json", () => {
+			const invalidJsonPayload = Buffer.from("not json", "utf8").toString(
+				"base64url",
+			);
+			const crypto = require("crypto");
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/web/src/lib/auth.test.ts]: Addressed by importing createHmac from crypto at the top of the ESM test file and removing the require call.

**Comment by is0692vs** [comment]
@greptile review

File: packages/web/src/lib/auth.test.ts [hunk 2]
```diff
@@ -1,199 +1,599 @@
...
+
+			const data = { auth: true };
+			const sealed = sealCookieValue(data);
+			const unsealed = unsealCookieValue(sealed);
+			expect(unsealed).toEqual(data);
+		});
+	});
+
+	describe("setAuthCookies", () => {
+		let mockResponse: any;
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/web/src/lib/auth.test.ts]: Addressed by replacing the any-typed mockResponse with a typed MockResponse helper shared by the cookie tests.

File: packages/web/src/lib/auth.test.ts [hunk 3]
```diff
@@ -1,199 +1,599 @@
...
+				cookies: {
+					set: vi.fn(),
+				},
+			};
+			vi.stubEnv("NODE_ENV", "development"); // to make isSecureRequest predictable
+			vi.stubEnv("COOKIE_SECRET", "super-secret-key-12345");
+		});
+
+		afterEach(() => {
+			vi.restoreAllMocks();
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/web/src/lib/auth.test.ts]: Addressed by switching the mocked-cookie test cleanup from vi.restoreAllMocks() to vi.resetAllMocks() while keeping environment cleanup intact.

File: packages/web/src/lib/auth.test.ts [hunk 4]
```diff
@@ -1,199 +1,599 @@
...
+
+		afterEach(() => {
+			vi.unstubAllEnvs();
+		});
+
+		it("should return null if cookie is not present", () => {
+			const cookieStore = {
+				get: vi.fn().mockReturnValue(undefined),
+			};
+			expect(getRefreshToken(cookieStore as any)).toBeNull();
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/web/src/lib/auth.test.ts]: Addressed by removing the cookieStore as any assertions in this test file and using the structural mock objects directly.

File: packages/web/src/lib/auth.test.ts [hunk 5]
```diff
@@ -1,199 +1,599 @@
...
+		});
+
+		it("should return null when the cookie is valid but contains no token", () => {
+			const sealed = sealCookieValue({ notToken: "abc" } as any);
+			const cookieStore = {
+				get: vi.fn().mockReturnValue({ value: sealed }),
+			};
+			expect(getRefreshToken(cookieStore as any)).toBeNull();
+		});
+	});
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/web/src/lib/auth.test.ts:378]: Addressed by fixing getRefreshToken to return only string tokens, plus adding the non-string token regression test.

File: packages/web/src/lib/auth.test.ts [hunk 6]
```diff
@@ -1,199 +1,599 @@
...
+				{
+					httpOnly: true,
+					secure: true,
+					sameSite: "lax",
+					path: "/",
+					maxAge: 600,
+				},
+			);
+			vi.unstubAllEnvs();
+		});
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/web/src/lib/auth.test.ts]: Addressed by replacing the undefined request fallback tests with real localhost and x-forwarded-proto request objects.

File: packages/web/src/lib/auth.test.ts [hunk 7]
```diff
@@ -1,199 +1,599 @@
...
+	unsealCookieValue,
+    getRefreshToken,
+    getSelectedDatabaseId,
+    isAuthenticated,
+    getNotionClient,
+    getUserInfo,
+    setDatabaseCookie,
+    setOauthStateCookie,
+    createStateToken
+} from "./auth";
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/web/src/lib/auth.test.ts:19]: Addressed by normalizing the auth.test.ts import block indentation to the file tab style.
